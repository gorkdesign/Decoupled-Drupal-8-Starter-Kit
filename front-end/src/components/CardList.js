import React from 'react';
import PropTypes from 'prop-types';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import { TransitionGroup, CSSTransition } from 'react-transition-group';

import MdAdd from 'react-icons/lib/md/add';
import Card from './Card';

const Fade = ({ children, ...props }) => (
  <CSSTransition
    {...props}
    timeout={1000}
    classNames="fade"
  >
    {children}
  </CSSTransition>
);
Fade.propTypes = {
  children: PropTypes.node.isRequired,
};

const listItems = ({ pages, ctaHandler, deleteItemHandler }) => pages.map(node => (
  <Fade duration={1000} key={node.nid} timeout={{ enter: 0, exit: 1000 }}>
    <Card
      node={node}
      ctaHandler={ctaHandler}
      deleteHandler={(event) => { deleteItemHandler(event, node.nid); }}
    />
  </Fade>
));

// listItems.PropTypes = {
//   nodes: PropTypes.arrayOf(PropTypes.shape({
//     author: PropTypes.shape({
//       name: PropTypes.string,
//     }),
//     body: {
//       value: PropTypes.string,
//     },
//     images: PropTypes.arrayOf(PropTypes.shape({
//       url: PropTypes.string,
//       mid: PropTypes.number,
//     })),
//     nid: PropTypes.number,
//     title: PropTypes.string,
//     uuid: PropTypes.string,
//   })).isRequired,
//   ctaHandler: PropTypes.func.isRequired,
//   deleteItemHandler: PropTypes.func.isRequired,
// };

const CardList = (props) => {
  const {
    addPageMutation, isModalVisible, onModalToggle, onModalOk,
  } = props;
  return (
    <div className="">
      <div className="container">
        <div
          role="button"
          tabIndex={0}
          onKeyUp={() => { addPageMutation('NULL'); }}
          className="py-3"
          onClick={() => { addPageMutation('NULL'); }}
        >
          <div className="add">
            <MdAdd />
            <h2 className="card-title">Add</h2>
          </div>
        </div>

        <TransitionGroup className="item-list">
          {listItems(props)}
        </TransitionGroup>

        <Modal isOpen={isModalVisible} toggle={onModalToggle} backdrop>
          <ModalHeader toggle={onModalToggle}>Confirmation</ModalHeader>
          <ModalBody>
            Are you sure you want to remove this?
          </ModalBody>
          <ModalFooter>
            <Button color="secondary" onClick={onModalToggle}>Cancel</Button>
            {' '}
            <Button color="primary" onClick={onModalOk}>Delete</Button>
          </ModalFooter>
        </Modal>
      </div>
    </div>
  );
};

CardList.propTypes = {
  isModalVisible: PropTypes.bool.isRequired,
  addPageMutation: PropTypes.func.isRequired,
  onModalToggle: PropTypes.func.isRequired,
  onModalOk: PropTypes.func.isRequired,
};

export default CardList;
