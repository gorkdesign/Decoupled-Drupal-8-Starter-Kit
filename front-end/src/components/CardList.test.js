import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import CardList from './CardList';

const props = {
  isModalVisible: false,
  nodes: [
    {
      author: "test",
      body:{
        value: "body test value"
      },
      images: [
        { 
          entity: {
            image: {
              derivative: {
                url: "http://d8d.loc/test.jpg",
                mid: 1
              }
            }
          }
        }
      ],
      nid: 34,
      title: "TEST TITLE",
      uuid: "09ba8da5-e08d-4972-a22b-ad7bb0a44d7f"
    },
    {
      author: "test2",
      body:{
        value: "body test value"
      },
      images: [
        { 
          entity: {
            image: {
              derivative: {
                url: "http://d8d.loc/test.jpg",
                mid: 1
              }
            }
          }
        }
      ],
      nid: 34,
      title: "TEST TITLE",
      uuid: "09ba8da5-e08d-4972-a22b-ad7bb0a44d7f"
    }
  ]
};

const generateShallowCardList = (props) => {
  return shallow(
    <CardList {...props}
      ctaHandler={jest.fn()} 
      deleteItemHandler={jest.fn()} 
      addPageMutation={jest.fn()} 
      onModalToggle={jest.fn()} 
      onModalOk={jest.fn()} 
    />
  );
}


describe('CardList', () => {
  it('should render correctly', () => {
    const output = generateShallowCardList(props);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});