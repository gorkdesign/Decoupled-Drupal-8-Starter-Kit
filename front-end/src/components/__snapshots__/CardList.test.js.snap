// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList should render correctly 1`] = `
<div
  className=""
>
  <div
    className="container"
  >
    <div
      className="py-3"
      onClick={[Function]}
    >
      <div
        className="add"
      >
        <MdAdd />
        <h2
          className="card-title"
        >
          Add
        </h2>
      </div>
    </div>
    <TransitionGroup
      childFactory={[Function]}
      className="item-list"
      component="div"
    >
      <Fade
        duration={1000}
        key="34"
        timeout={
          Object {
            "enter": 0,
            "exit": 1000,
          }
        }
      >
        <Card
          ctaHandler={[Function]}
          deleteHandler={[Function]}
          node={
            Object {
              "author": "test",
              "body": Object {
                "value": "body test value",
              },
              "images": Array [
                Object {
                  "entity": Object {
                    "image": Object {
                      "derivative": Object {
                        "mid": 1,
                        "url": "http://d8d.loc/test.jpg",
                      },
                    },
                  },
                },
              ],
              "nid": 34,
              "title": "TEST TITLE",
              "uuid": "09ba8da5-e08d-4972-a22b-ad7bb0a44d7f",
            }
          }
        />
      </Fade>
      <Fade
        duration={1000}
        key="34"
        timeout={
          Object {
            "enter": 0,
            "exit": 1000,
          }
        }
      >
        <Card
          ctaHandler={[Function]}
          deleteHandler={[Function]}
          node={
            Object {
              "author": "test2",
              "body": Object {
                "value": "body test value",
              },
              "images": Array [
                Object {
                  "entity": Object {
                    "image": Object {
                      "derivative": Object {
                        "mid": 1,
                        "url": "http://d8d.loc/test.jpg",
                      },
                    },
                  },
                },
              ],
              "nid": 34,
              "title": "TEST TITLE",
              "uuid": "09ba8da5-e08d-4972-a22b-ad7bb0a44d7f",
            }
          }
        />
      </Fade>
    </TransitionGroup>
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropTransition={
        Object {
          "mountOnEnter": true,
          "timeout": 150,
        }
      }
      fade={true}
      isOpen={false}
      keyboard={true}
      modalTransition={
        Object {
          "timeout": 300,
        }
      }
      onClosed={[Function]}
      onOpened={[Function]}
      role="dialog"
      toggle={[Function]}
      zIndex={1050}
    >
      <ModalHeader
        closeAriaLabel="Close"
        tag="h5"
        toggle={[Function]}
        wrapTag="div"
      >
        Confirmation
      </ModalHeader>
      <ModalBody
        tag="div"
      >
        Are you sure you want to remove this?
      </ModalBody>
      <ModalFooter
        tag="div"
      >
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
        >
          Cancel
        </Button>
         
        <Button
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          Delete
        </Button>
      </ModalFooter>
    </Modal>
  </div>
</div>
`;
